plugins/openai/composio_openai/toolset.py

```diff
@@ -0,0 +1,233 @@
+"""
+OpenAI tool spec.
+"""
+
+import json
+import time
+import typing as t
+
+from openai import Client
+from openai.types.beta.thread import Thread
+from openai.types.beta.threads.run import RequiredAction, Run
+from openai.types.chat.chat_completion import ChatCompletion
+from openai.types.chat.chat_completion_message_tool_call import (
+    ChatCompletionMessageToolCall,
+)
+from openai.types.chat.chat_completion_tool_param import ChatCompletionToolParam
+
+from composio.client.enums import Action, App, Tag
+from composio.constants import DEFAULT_ENTITY_ID
+from composio.tools import ComposioToolSet as BaseComposioToolSet
+from composio.tools.schema import OpenAISchema, SchemaType
+
+
+class ComposioToolSet(BaseComposioToolSet):
+    """
+    Composio toolset for OpenAI framework.
+
+    Example:
+    ```python
+        import dotenv
+        from composio_openai import App, ComposioToolset
+        from openai import OpenAI
+
+
+        # Load environment variables from .env
+        dotenv.load_dotenv()
+
+        # Initialize tools.
+        openai_client = OpenAI()
+        composio_tools = ComposioToolset()
+
+        # Define task.
+        task = "Star a repo SamparkAI/composio_sdk on GitHub"
+
+        # Get GitHub tools that are pre-configured
+        actions = composio_toolset.get_tools(tools=[App.GITHUB])
```

Comment:
Seems incorrect. It is apps=[App.Github] and not tools. May be we rename the function to get_apps??